.github/workflows/build_and_test.yml

```diff
@@ -16,43 +16,82 @@ on:
+  # pytket-braket
+  PYTKET_REMOTE_BRAKET_ACCESS_KEY_ID: ${{ secrets.PYTKET_REMOTE_BRAKET_ACCESS_KEY_ID }}
+  PYTKET_REMOTE_BRAKET_ACCESS_KEY_SECRET: ${{ secrets.PYTKET_REMOTE_BRAKET_ACCESS_KEY_SECRET }}
+  PYTKET_REMOTE_BRAKET_REGION: ${{ secrets.PYTKET_REMOTE_BRAKET_REGION }}
+  PYTKET_REMOTE_BRAKET_BUCKET: ${{ secrets.PYTKET_REMOTE_BRAKET_BUCKET }}
+  PYTKET_REMOTE_BRAKET_FOLDER: ${{ secrets.PYTKET_REMOTE_BRAKET_FOLDER }}
+  # pytket-iqm
+  PYTKET_REMOTE_IQM_USERNAME: ${{ secrets.PYTKET_REMOTE_IQM_USERNAME }}
+  PYTKET_REMOTE_IQM_APIKEY: ${{ secrets.PYTKET_REMOTE_IQM_APIKEY }}
+  # pytket-qiskit
+  PYTKET_REMOTE_QISKIT_TOKEN: ${{ secrets.PYTKET_REMOTE_QISKIT_TOKEN }}
+  # pytket-qsharp
+  PYTKET_REMOTE_QSHARP_RESOURCE_ID: ${{ secrets.PYTKET_REMOTE_QSHARP_RESOURCE_ID }}
+  PYTKET_REMOTE_QSHARP_LOCATION: ${{ secrets.PYTKET_REMOTE_QSHARP_LOCATION }}
+  PYTKET_REMOTE_QSHARP_STORAGE: ${{ secrets.PYTKET_REMOTE_QSHARP_STORAGE }}
+  AZURE_TENANT_ID: ${{ secrets.PYTKET_REMOTE_QSHARP_AZURE_TENANT_ID }}
+  AZURE_CLIENT_ID: ${{ secrets.PYTKET_REMOTE_QSHARP_AZURE_CLIENT_ID }}
+  AZURE_CLIENT_SECRET: ${{ secrets.PYTKET_REMOTE_QSHARP_AZURE_CLIENT_SECRET }}
+  # pytket-quantinuum
+  PYTKET_REMOTE_QUANTINUUM_USERNAME: ${{ secrets.PYTKET_REMOTE_QUANTINUUM_USERNAME }}
+  PYTKET_REMOTE_QUANTINUUM_PASSWORD: ${{ secrets.PYTKET_REMOTE_QUANTINUUM_PASSWORD }}
 
 jobs:
+  set-all-modules-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/* +refs/heads/*:refs/remotes/origin/*
+      - name: Get json list of all the pytket extensions
+        id: set-all-modules-matrix
+        run: |
+          sudo apt-get install jq
+          cd ${GITHUB_WORKSPACE}/modules
+          LS_MODULES=`ls -d */`
+          ALL_MODULES=`echo ${LS_MODULES} | sed -e 's/\///g'`
+          echo "MODULES_TO_TEST=${ALL_MODULES}" >> $GITHUB_ENV
+          ALL_MODULES_LIST=($ALL_MODULES)
+          echo "::set-output name=all-modules::$(printf '%s\n' ${ALL_MODULES_LIST[@]} | jq -R . | jq -cs .)"
+    outputs:
+      matrix: ${{ steps.set-all-modules-matrix.outputs.all-modules }}
 
   linux:
     name: Build and test (Linux)
+    needs: set-all-modules-matrix
     runs-on: ubuntu-20.04
 
-    steps:
+    strategy:
+      matrix:
+        module: ${{ fromJson(needs.set-all-modules-matrix.outputs.matrix) }}
+      fail-fast: false
 
+    steps:
     - uses: actions/checkout@v2
       with:
         fetch-depth: '0'
     - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/* +refs/heads/*:refs/remotes/origin/*
-    - name: Initialize modules list
-      run: |
-        cd ${GITHUB_WORKSPACE}/modules
-        LS_MODULES=`ls -d */`
-        ALL_MODULES=`echo ${LS_MODULES} | sed -e 's/\///g'`
-        echo "MODULES_TO_TEST=${ALL_MODULES}" >> $GITHUB_ENV
-    - name: check for mofified modules
+    - name: Only test modified modules for PRs
```

Comment:
I don't think this is working. For instance, in this PR we see pytket-aqt being tested, even though it is unmodified. We seem to be setting `MODULES_TO_TEST` in the environment to `matrix.module`, overwriting this value.